backend/portland_ingest.py

```diff
@@ -0,0 +1,151 @@
+
+        spring_data_sat["season"] = "Spring"
+        spring_data_week["season"] = "Spring"
+        spring_data_sun["season"] = "Spring"
+
+        summer_data_sat["season"] = "Summer"
+        summer_data_week["season"] = "Summer"
+        summer_data_sun["season"] = "Summer"
+
+        # append into one data set
+        all_rider_data = pd.concat(
+            [
+                spring_data_sat,
+                spring_data_week,
+                spring_data_sun,
+                summer_data_sat,
+                summer_data_week,
+                summer_data_sun,
+            ],
+            ignore_index=True,
+        )
+
+        print("Data loaded and appended successfully.")
+
+    except FileNotFoundError:
+        print("Files not found. Check the file path and filename.")
+    except Exception as e:
+        print("An error occurred:", e)
+
+
+# load data sets
+all_rider_data = load_ridership_dasets()
+
+# format data sets
+all_rider_data.columns = all_rider_data.columns.str.lower().str.replace(" ", "_")
+
+all_rider_data = all_rider_data[["location_id", "ons", "day_type", "season"]]
+
+# Since the data is at the day type + season level,
+# create a df with all calendar days of 2023
+start_date = "2023-01-01"
+end_date = "2023-12-31"
+calendar_dates = pd.date_range(start=start_date, end=end_date, freq="D")
+
+# flag days of 2023 to match day_type field
+calendar_df = pd.DataFrame(calendar_dates, columns=["date"])
+
+calendar_df["day_type"] = calendar_df["date"].dt.day_name()
+
+day_type_mapping = {"Saturday": "Saturday", "Sunday": "Sunday"}
+
+calendar_df["day_type"] = (
+    calendar_df["day_type"].map(day_type_mapping).fillna("Weekday")
+)
+
+# flag seasons, all days are flagged to Spring, except for summer months,
+# to differenciate for routes that only run during the summer months
+calendar_df["season"] = "Spring"
+
+# Summer 2023 date range
+summer_start = pd.to_datetime("2023-06-21")
+summer_end = pd.to_datetime("2023-09-22")
+
+calendar_df.loc[
+    (calendar_df["date"] >= summer_start) & (calendar_df["date"] <= summer_end),
+    "season",
+] = "Summer"
+
+# combine Tri Met data with calendar dates in order to convert the data set
+# from day_type (sat, sun, or weekday) + season level into date level
+
+merged_data = pd.merge(
+    all_rider_data, calendar_df, on=["day_type", "season"], how="left"
+)
+
+sorted_data = merged_data.sort_values(by=["date", "day_type", "season"], ascending=True)
+
+sorted_data = sorted_data.reindex(columns=["location_id", "date", "ons"])
+
+sorted_data.rename(columns={"ons": "riders"}, inplace=True)
+
+sorted_data["date"] = pd.to_datetime(sorted_data["date"])
+
+# modify location_id to include city code
+sorted_data["location_id"] = sorted_data["location_id"].apply(lambda x: f"Portland-{x}")
+
+# export final data set in JSON format
+output_file = folder_path + "Portland_ridership.json"
+sorted_data.to_json(
+    output_file,
+    orient="records",
+)
```

Comment:
we'll likely want to refactor this into:
1) more functions to reduce repeated calls of very similar actions
2) make it easy to change which time frame/data is being called so this will continue to work as time goes on (instead of just 2023) 
3) move everything into functions so that it can be used in other scripts easily as we aggregate data together (line 90 onwards)

but these can likely be done when the ingestion pipeline is being finalized.